README.md

```diff
@@ -1,2 +1,220 @@
+    the user for authentication. Upon successful authentication, the request would engage the 
+    **HTTPSessionBasedOIDCProcessor** and the user would be redirected to the **home.jsp** page.
+
+### Retrieve user attributes
+
+The web app needs to be configured to read the attributes sent from the Identity Server upon successful
+ authentication. In the oidc-sample-app, we would customize the home.jsp file as follows to retrieve the user
+  attributes.
+ 
+ ```
+<%
+    // Retrieve the current session.
+    final HttpSession currentSession = request.getSession(false);
+
+    // Logged in session context.
+    final SessionContext sessionContext = (SessionContext)
+            currentSession.getAttribute(SSOAgentConstants.SESSION_CONTEXT);
+
+    // Logged in user.
+    final User user = sessionContext.getUser();
+
+    // Attributes of the logged in user.
+    Map<String, Object> customClaimValueMap = user.getAttributes();
+%>
+```
+
+
+### Trigger logout
+
+In the **home.jsp** file, we have added the following to trigger a logout flow:
+
+``<a href='logout'>Logout</a>``
+
+Clicking on the logout link would trigger the logout flow engaging the same **OIDCAgentFilter** mentioned above.
+After successful logout, the user would be redirected to the page configured via the `indexPage` property previously
+ discussed.
+
+## Integrating Asgardio Tomcat OIDC Agent
+
+Asgardio Tomcat OIDC Agent can be integrated in to your applications in two different ways. 
+
+It can be integrated to your java source project of the webapp when the web application is in development stage.
+
+And, the Tomcat OIDC agent can be integrated into a pre-built webapp as well.
+
+#### To your existing webapp
+
+To integrate the Tomcat OIDC Agent into your pre-built webapps, follow the guide [here](docs/integrating_with_existing_webapp.md/#Integrating_OIDC_into_your_existing_Webapp).
+
+#### To your Java source project
+
+To integrate the Tomcat OIDC Agent into your java source project, follow the guide [here](docs/integrating_with_java_source_project.md/#Integrating_OIDC_into_your_java_source_project).
+
+
+## Installing the Agent [TODO]
+
+### Github
+The SDK is hosted on github. You can download it from:
+- Latest release: [TODO]
+- Master repo: [TODO]
+
+### Building from the source
+
+If you want to build **asgardio-tomcat-oidc-agent** from the source code:
+
+1. Install Java 8
+2. Install Apache Maven 3.x.x (https://maven.apache.org/download.cgi#)
+3. Get a clone or download the source from this repository (https://github.com/asgardio/asgardio-tomcat-oidc-agent.git)
+4. Run the Maven command ``mvn clean install`` from the ``asgardio-tomcat-oidc-agent`` directory.
+
+### Maven
+
+Install it as a maven dependency:
+```
+<dependency>
+    <groupId>io.asgardio.tomcat.oidc.agent</groupId>
+    <artifactId>io.asgardio.tomcat.oidc.agent</artifactId>
+    <version>1.0.0</version>
```

Comment:
0.1.0-SNAPSHOT

Reply:
Resolved in https://github.com/asgardio/asgardio-tomcat-oidc-agent/pull/7/commits/bf4f7fa7367a16bd78c9cbabc954df5fb7451e18